### PR TITLE
Fix MR "Add support for NTLMv2 (#551)".

### PR DESCRIPTION
**Pull request by NikitaDef**
Domain has to be used from connection string, the user domain was used from the remote server.
The user domain sensitive to the case which is defined in the connection string, the fix can break some user apps, which has set domain wrong case before.

I use https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-nlmp/5e550938-91d4-459f-b67d-75d70009e3f3

### CODE REVIEW

File: ntlm.go [hunk 1]
```diff
@@ -251,18 +251,18 @@ func getNTLMv2AndLMv2ResponsePayloads(target, username, password string, challen
...
 
-func negotiateExtendedSessionSecurity(flags uint32, message []byte, challenge [8]byte, username, password string) (lm, nt []byte, err error) {
+func negotiateExtendedSessionSecurity(flags uint32, message []byte, challenge [8]byte, username, password, userDom string) (lm, nt []byte, err error) {
 	nonce := clientChallenge()
 
 	// Official specification: https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-nlmp/b38c36ed-2804-4868-a9ff-8dd3182128e4
 	// Unofficial walk through referenced by https://www.freetds.org/userguide/domains.htm: http://davenport.sourceforge.net/ntlm.html
 	if (flags & _NEGOTIATE_TARGET_INFO) != 0 {
-		targetInfoFields, target, err := getNTLMv2TargetInfoFields(message)
+		targetInfoFields, _, err := getNTLMv2TargetInfoFields(message)
```
Comment (kardianos) [inline ntlm.go:260]: With this change, the target from this message isn't used anywhere. That may be fine. From what I can tell, you are substituting the domain in the username rather then using the ntlm derived authentication context. Is that somewhat correct?

Comment (NikitaDef) [reply ntlm.go:260]: Oh, I forgot remove it. I'll fix it

**Comment by kardianos** [comment]
LGTM, waiting for user feedback on issue. Will merge if no feedback.

**Comment by sboutzen** [comment]
Nice find!

**Pull request merged**